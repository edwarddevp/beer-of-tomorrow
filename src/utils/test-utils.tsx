import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "../theme";
import { createSerializer, matchers } from "@emotion/jest";
import { useStore } from "@/configs/store";
import { Provider } from "react-redux";
import mockBeerItems from "./mock-beer-items";
// @ts-ignore
expect.addSnapshotSerializer(createSerializer());
// @ts-ignore
expect.extend(matchers);

const initialStoreValues = {
  getBeerById: {
    data: mockBeerItems[0],
    isLoading: false,
    errors: [],
  },
};

const AllTheProviders: FC = ({ children }) => {
  const store = useStore(initialStoreValues);
  return (
    <ChakraProvider theme={theme}>
      <Provider store={store}>{children}</Provider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
