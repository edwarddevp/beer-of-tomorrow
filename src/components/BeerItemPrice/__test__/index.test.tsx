import mockBeerItems from "@/utils/mock-beer-items";
import React from "react";
// Using render and screen from test-utils.js instead of
// @testing-library/react
import { render } from "test-utils";
import { BeerItemPrice } from "../";

describe("BeerItemPrice", () => {
  it("renders", () => {
    const { container } = render(
      <BeerItemPrice beer={mockBeerItems[0]} isInOffert={false} />
    );
    expect(container).toMatchSnapshot();
  });
  it("renders isInOffer", () => {
    const { container } = render(
      <BeerItemPrice beer={mockBeerItems[0]} isInOffert={true} />
    );
    expect(container).toMatchSnapshot();
  });
  it("If item doesn't have a price", () => {
    const { container } = render(<BeerItemPrice beer={{}} isInOffert={true} />);
    expect(container).toMatchSnapshot();
  });
});
