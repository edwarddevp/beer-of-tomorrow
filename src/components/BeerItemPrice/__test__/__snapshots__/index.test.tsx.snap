// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerItemPrice If item doesn't have a price 1`] = `<div />`;

exports[`BeerItemPrice renders 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-1 {
  color: var(--chakra-colors-black);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-4xl);
}

.emotion-2 {
  color: var(--chakra-colors-gray-500);
  font-size: var(--chakra-fontSizes-lg);
}

@media screen and (min-width: 30em) {
  .emotion-2 {
    font-size: var(--chakra-fontSizes-xl);
  }
}

<div>
  <div
    class="chakra-stack emotion-0"
  >
    <p
      class="chakra-text emotion-1"
    >
      $
      1044
    </p>
    <p
      class="chakra-text emotion-2"
    >
      $ 
      20
    </p>
  </div>
</div>
`;

exports[`BeerItemPrice renders isInOffer 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: var(--chakra-space-2);
  margin-inline-start: var(--chakra-space-2);
}

.emotion-2 {
  color: var(--chakra-colors-primary-500);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-4xl);
}

.emotion-3 {
  font-weight: var(--chakra-fontWeights-bold);
  color: var(--chakra-colors-gray-300);
  font-size: var(--chakra-fontSizes-sm);
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  padding-top: var(--chakra-space-2);
}

@media screen and (min-width: 30em) {
  .emotion-3 {
    font-size: var(--chakra-fontSizes-xl);
  }
}

.emotion-4 {
  color: var(--chakra-colors-gray-500);
  font-size: var(--chakra-fontSizes-lg);
}

@media screen and (min-width: 30em) {
  .emotion-4 {
    font-size: var(--chakra-fontSizes-xl);
  }
}

<div>
  <div
    class="chakra-stack emotion-0"
  >
    <div
      class="chakra-stack emotion-1"
    >
      <p
        class="chakra-text emotion-2"
      >
        $
        1044
      </p>
      <p
        class="chakra-text emotion-3"
      >
        $
        1012.68
      </p>
    </div>
    <p
      class="chakra-text emotion-4"
    >
      $ 
      20
    </p>
  </div>
</div>
`;
